Source/Urho3D/Scene/Node.cpp

```diff
@@ -495,11 +498,41 @@ void Node::SetTransform(const Vector3& position, const Quaternion& rotation, flo
 
 void Node::SetTransform(const Vector3& position, const Quaternion& rotation, const Vector3& scale)
 {
+
+    Vector3 oldPosition = position_;
+    Quaternion oldRotation = rotation_;
+    Vector3 oldScale = scale_;
+    Matrix3x4 oldTransform = GetTransform();
+
     position_ = position;
     rotation_ = rotation;
     scale_ = scale;
-    MarkDirty();
 
+    if (sendTransformEvents_) {

```

Comment:
physics world is ground truth.  Except in the case that you explicity move a node with the API. in that case the physics world moves the associated rigid body.  I take the stance of if you move a node you move a node.


Other than that the new event is not necessarily a thing related to physics anyway. Just a useful event to hook to.

Reply:
This is wrong. `RigidBody` is supposed to be manipulated in this case. You can clearly feel consequences of wrong assumption as you had to modify renderer. This should not be necessary. Physics engine integration should be confined to modifications in Physics subfolder. If you need to touch something outside - it is a clear red flag something is going off the rails.

Reply:
another approach might be that the physics component could monitor a dirty flag on node indicating a transform change was made explicitly.  I actually like that idea better than the event because event could be called alot unnecessarily.  But I do want SetTransform() to alter the physics world.

Reply:
I guess your convincing me on this one.  Maybe rigidbody could have a method like "SnapToNodeTransform()"  then these node changes could be removed.  I would be happy with that.

Reply:
There already is `RigidBody::SetTransform()`. I wonder why you do not follow existing API.